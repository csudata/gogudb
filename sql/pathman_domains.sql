\set VERBOSITY terse

SET search_path = 'public','_gogu';
CREATE EXTENSION gogudb;
CREATE SCHEMA domains;

CREATE DOMAIN domains.dom_test AS numeric CHECK (value < 1200);

CREATE TABLE domains.dom_table(val domains.dom_test NOT NULL);
INSERT INTO domains.dom_table SELECT generate_series(1, 999);

SELECT create_range_partitions('domains.dom_table', 'val', 1, 100);

EXPLAIN (COSTS OFF)
SELECT * FROM domains.dom_table
WHERE val < 250;

INSERT INTO domains.dom_table VALUES(1500);
INSERT INTO domains.dom_table VALUES(-10);

SELECT append_range_partition('domains.dom_table');
SELECT prepend_range_partition('domains.dom_table');
SELECT merge_range_partitions('gogudb_partition_table._domains_1_dom_table', 'gogudb_partition_table._domains_2_dom_table');
SELECT split_range_partition('gogudb_partition_table._domains_1_dom_table', 50);

INSERT INTO domains.dom_table VALUES(1101);

EXPLAIN (COSTS OFF)
SELECT * FROM domains.dom_table
WHERE val < 450;


SELECT * FROM gogudb_partition_list
ORDER BY range_min::INT, range_max::INT;


SELECT drop_partitions('domains.dom_table');
SELECT create_hash_partitions('domains.dom_table', 'val', 5);

SELECT * FROM gogudb_partition_list
ORDER BY "partition"::TEXT;


DROP SCHEMA domains CASCADE;
DROP EXTENSION gogudb CASCADE;

