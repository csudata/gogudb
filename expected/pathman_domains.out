\set VERBOSITY terse
SET search_path = 'public','_gogu';
CREATE EXTENSION gogudb;
CREATE SCHEMA domains;
CREATE DOMAIN domains.dom_test AS numeric CHECK (value < 1200);
CREATE TABLE domains.dom_table(val domains.dom_test NOT NULL);
INSERT INTO domains.dom_table SELECT generate_series(1, 999);
SELECT create_range_partitions('domains.dom_table', 'val', 1, 100);
 create_range_partitions 
-------------------------
                      10
(1 row)

EXPLAIN (COSTS OFF)
SELECT * FROM domains.dom_table
WHERE val < 250;
                    QUERY PLAN                     
---------------------------------------------------
 Append
   ->  Seq Scan on _domains_1_dom_table
   ->  Seq Scan on _domains_2_dom_table
   ->  Seq Scan on _domains_3_dom_table
         Filter: ((val)::numeric < '250'::numeric)
(5 rows)

INSERT INTO domains.dom_table VALUES(1500);
ERROR:  value for domain domains.dom_test violates check constraint "dom_test_check"
INSERT INTO domains.dom_table VALUES(-10);
SELECT append_range_partition('domains.dom_table');
            append_range_partition            
----------------------------------------------
 gogudb_partition_table._domains_12_dom_table
(1 row)

SELECT prepend_range_partition('domains.dom_table');
           prepend_range_partition            
----------------------------------------------
 gogudb_partition_table._domains_13_dom_table
(1 row)

SELECT merge_range_partitions('gogudb_partition_table._domains_1_dom_table', 'gogudb_partition_table._domains_2_dom_table');
 merge_range_partitions 
------------------------
 
(1 row)

SELECT split_range_partition('gogudb_partition_table._domains_1_dom_table', 50);
 split_range_partition 
-----------------------
 {1,201}
(1 row)

INSERT INTO domains.dom_table VALUES(1101);
EXPLAIN (COSTS OFF)
SELECT * FROM domains.dom_table
WHERE val < 450;
                    QUERY PLAN                     
---------------------------------------------------
 Append
   ->  Seq Scan on _domains_13_dom_table
   ->  Seq Scan on _domains_11_dom_table
   ->  Seq Scan on _domains_1_dom_table
   ->  Seq Scan on _domains_14_dom_table
   ->  Seq Scan on _domains_3_dom_table
   ->  Seq Scan on _domains_4_dom_table
   ->  Seq Scan on _domains_5_dom_table
         Filter: ((val)::numeric < '450'::numeric)
(9 rows)

SELECT * FROM gogudb_partition_list
ORDER BY range_min::INT, range_max::INT;
      parent       |                  partition                   | parttype | expr | range_min | range_max 
-------------------+----------------------------------------------+----------+------+-----------+-----------
 domains.dom_table | gogudb_partition_table._domains_13_dom_table |        2 | val  | -199      | -99
 domains.dom_table | gogudb_partition_table._domains_11_dom_table |        2 | val  | -99       | 1
 domains.dom_table | gogudb_partition_table._domains_1_dom_table  |        2 | val  | 1         | 50
 domains.dom_table | gogudb_partition_table._domains_14_dom_table |        2 | val  | 50        | 201
 domains.dom_table | gogudb_partition_table._domains_3_dom_table  |        2 | val  | 201       | 301
 domains.dom_table | gogudb_partition_table._domains_4_dom_table  |        2 | val  | 301       | 401
 domains.dom_table | gogudb_partition_table._domains_5_dom_table  |        2 | val  | 401       | 501
 domains.dom_table | gogudb_partition_table._domains_6_dom_table  |        2 | val  | 501       | 601
 domains.dom_table | gogudb_partition_table._domains_7_dom_table  |        2 | val  | 601       | 701
 domains.dom_table | gogudb_partition_table._domains_8_dom_table  |        2 | val  | 701       | 801
 domains.dom_table | gogudb_partition_table._domains_9_dom_table  |        2 | val  | 801       | 901
 domains.dom_table | gogudb_partition_table._domains_10_dom_table |        2 | val  | 901       | 1001
 domains.dom_table | gogudb_partition_table._domains_12_dom_table |        2 | val  | 1001      | 1101
 domains.dom_table | gogudb_partition_table._domains_15_dom_table |        2 | val  | 1101      | 1201
(14 rows)

SELECT drop_partitions('domains.dom_table');
NOTICE:  49 rows copied from gogudb_partition_table._domains_1_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_3_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_4_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_5_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_6_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_7_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_8_dom_table
NOTICE:  100 rows copied from gogudb_partition_table._domains_9_dom_table
NOTICE:  99 rows copied from gogudb_partition_table._domains_10_dom_table
NOTICE:  1 rows copied from gogudb_partition_table._domains_11_dom_table
NOTICE:  0 rows copied from gogudb_partition_table._domains_12_dom_table
NOTICE:  0 rows copied from gogudb_partition_table._domains_13_dom_table
NOTICE:  151 rows copied from gogudb_partition_table._domains_14_dom_table
NOTICE:  1 rows copied from gogudb_partition_table._domains_15_dom_table
 drop_partitions 
-----------------
              14
(1 row)

SELECT create_hash_partitions('domains.dom_table', 'val', 5);
 create_hash_partitions 
------------------------
                      5
(1 row)

SELECT * FROM gogudb_partition_list
ORDER BY "partition"::TEXT;
      parent       |                  partition                  | parttype | expr | range_min | range_max 
-------------------+---------------------------------------------+----------+------+-----------+-----------
 domains.dom_table | gogudb_partition_table._domains_0_dom_table |        1 | val  |           | 
 domains.dom_table | gogudb_partition_table._domains_1_dom_table |        1 | val  |           | 
 domains.dom_table | gogudb_partition_table._domains_2_dom_table |        1 | val  |           | 
 domains.dom_table | gogudb_partition_table._domains_3_dom_table |        1 | val  |           | 
 domains.dom_table | gogudb_partition_table._domains_4_dom_table |        1 | val  |           | 
(5 rows)

DROP SCHEMA domains CASCADE;
NOTICE:  drop cascades to 7 other objects
DROP EXTENSION gogudb CASCADE;
